imap/ical_support.c

```diff
@@ -3241,6 +3241,44 @@ EXPORTED void icalcomponent_set_jmapid(icalcomponent *comp, const char *id)
     icalproperty_set_x_name(prop, JMAPICAL_XPROP_ID);
     icalcomponent_add_property(comp, prop);
 }
+
+EXPORTED char *icalparameter_get_sanitized_cn_r(icalparameter *param)
+{
+    const char *cn = icalparameter_get_cn(param);
+    if (!cn) return NULL;
+
+    struct buf buf = BUF_INITIALIZER;
+
+    int has_specials = 0;
+    for (const char *c = cn; *c; c++) {
+        if (strchr(MIME_TSPECIALS, *c)) {
+            has_specials = 1;
+            break;
+        }
+    }
+
+    if (has_specials) {
+        struct address *a = NULL;
```

Comment:
The has_specials loop is correct, but it just occurred to me that the loop and has_specials variable could all be replaced by just:

`if (strcspn(cn, MIME_SPECIALS) < strlen(cn))`

Reply:
I like it, thanks :) Updated accordingly.